src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -129,6 +129,18 @@
       <SourcesToCopyToTestDir Include="@(OverridenRuntimeContent)" />
       <DestinationsForTestDir Include="@(OverridenRuntimeContent->'$(TestPath)\%(Filename)%(Extension)')" />
     </ItemGroup>
+    
+    <!-- Removing assets from TestCopyLocal that come in from P2P references in SourcesToCopyToTestDir -->
+    <ItemGroup>
+        <_FileNamesInSourcesToCopyToTestDir Include="@(SourcesToCopyToTestDir->'%(Filename)%(Extension)')">
+          <SourcePath>%(SourcesToCopyToTestDir.Identity)</SourcePath>
+        </_FileNamesInSourcesToCopyToTestDir>
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
+          <SourcePath>%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)</SourcePath>
```

Comment:
Moreover I don't think you need this item.  I believe _TestCopyLocalByFileNameWithoutDuplicates already has Identity of FileName+Extension and has SourcePath metadata.